src/testFixtures/java/org/opensearch/neuralsearch/BaseNeuralSearchIT.java

```diff
@@ -146,6 +147,7 @@ public abstract class BaseNeuralSearchIT extends OpenSearchSecureRestTestCase {
 
     protected ThreadPool threadPool;
     protected ClusterService clusterService;
+    private static final Set<String> DEPLOYED_MODEL_IDS = new HashSet<>();
```

Comment:
use concurrent safe implementation


```suggestion
    private static final Set<String> DEPLOYED_MODEL_IDS = ConcurrentHashMap.newKeySet();
```

Reply:
also clean up this set in any of the "afterclass" (or similar) methods